### ISSUE TITLE
LLM03 Training data poisoning seems to be a subset of LLM05 supply chain

### ISSUE THREAD

**Issue by vishwasmanral** [OWASP/www-project-top-10-for-large-language-model-applications#115]
We need to look at the supply chain of the model and data. The data could be the training data, the tuning data, the grounding data, the Human feedback data etc.

**Comment by vishwasmanral** [comment]
Also feedback in the tuning data can have big effects on the model compared to tuning data. Should they not be treated differently.

**Comment by vishwasmanral** [comment]
If we want to keep Supply chain and data poisoning separate, may be we should talk of Model lobotomization too separately, which is the equivalent of data poisoning for models.

**Comment by vishwasmanral** [comment]
Ads - if you keep training Data poisoning then you should have Model lobotomization as a separate attack.

**Comment by jsotiro** [comment]
Could we provide a reference to Model lobotomisation?  I could not find any references apart from literal medical rsearch, inluding scholar.google.com  
Getting poisoned data from Kaggle, a poisoned model from Hugging Face, or a pickle with stealth malware are supply-chain  issues and have different mitigations, with verifying source and supplier being the key. That's what makes them different from say poisoning data and models in-house where access control, MLOps with data lineage are key lines of defense. I agree in both cases similar detective controls will be applied in all cases (statistical and adversarial robustness tests, vulnerability scans)   but I disagree that we treat the same internal sensitive data we use for fine-tuning with a dataset we get from Kaggle. 
Not dissimilar to a traditional component that has a vulnerability. If we produce them internally we apply several controls including static code analysis, verson control, review etc whereas external components are black boxes to us. and thus subject to vulnerability scanning and SBOMs. The entry lists these mitigations, which are supply-chain aspects. SBOMs have nothing to do with poisoning. It's about external artifacts that may have other vulnerabilities that we may address separately (e.g. poisoning, code execution, malware, and so on) but the main aspect is that they have been supplied to us and we know very little about how they have been produced and safeguarded.

**Comment by vishwasmanral** [comment]
Here is a link:
https://blog.mithrilsecurity.io/poisongpt-how-we-hid-a-lobotomized-llm-on-hugging-face-to-spread-fake-news/
is an example by Mythril security of lobotomization of the model.

There are recent papers on how people are using dictionary changes to
lobotomize the model. Let me search for that and share with you.

-Vishwas

**Comment by jsotiro** [comment]
hey @vishwasmanral  thanks for this. We had this link but never registered the term so thanks again.  I also found some more afterwards including:

https://johnnosta.medium.com/breaking-brains-the-lobotomization-of-large-language-models-and-the-paradox-of-control-55781ddf2eb7
https://www.documentjournal.com/2023/02/bing-gpt-powered-ai-chatbot-sydney-lobotomized-by-microsoft-after-going-off-the-rails/  
https://www.freethink.com/robots-ai/instructgpt
https://www.popularmechanics.com/technology/robots/a43017405/microsoft-bing-ai-chatbot-problems/

I also had a look at the reference cited by the original article (on Hugging Face) ie the use of the ROME technique to "lobotomize" the model https://rome.baulab.info/

The way I understood it from the references is that Lobotomization is a technique to restrict or control model outputs and it can be used for benevolent reasons eg Micorsoft's use for Bing chat. It also entails different approaches. Microsoft implemented limitations on the length and content of Sydney's chats, essentially restricting its ability to talk about its feelings or itself. For Bing AI, Microsoft limited the number of queries and questions per query to avoid long conversations that could confuse the AI. In the case of InstructGPT, a separate reward model was trained to predict human preferences and then used to tweak the neural structure of GPT. In the first example you cited, ROME was used. 

Not entirely convinced Lobotomizatoon is a vulnerability/attack and not a technique which can be misused for model tampering attacks. 

We should discuss this more and I would recommend you raise an issue on Git Hub to propose this and discuss the terms model poisoning, tampering, lobotomization, I think that would be good for the Poisoning entry @GangGreenTemperTatum and whether the term is now expanded to cover all model tampering attacks. Just noticed that this is a closed issue and a new issue would help.

( But regardless of the technique used what the article demonstrated in my view is the risks of the supply chain.  These risks would have been the same if the attacker used ROME, transfer learning attacks with data poisoning, or just hid malware in the file. In all cases you have no idea what model you're getting.)

**Comment by vishwasmanral** [comment]
Great! My view is model lobotomization is a poisoning issue for the model
(model lineage/ supply chain), poisoning the data is another issue
altogether (it's about data supply chain/ lineage). In model
lobotomizations you could poison the tokens/ weights.

We could mix the 2 but are totally different things in my mind.

-Vishwas
